lib/puppet/network/http/api.rb

```diff
@@ -1,9 +1,11 @@
 class Puppet::Network::HTTP::API
+
   def self.not_found
     Puppet::Network::HTTP::Route.
       path(/.*/).
       any(lambda do |req, res|
-        raise Puppet::Network::HTTP::Error::HTTPNotFoundError.new("No route for #{req.method} #{req.path}", Puppet::Network::HTTP::Issues::HANDLER_NOT_FOUND)
+        raise Puppet::Network::HTTP::Error::HTTPNotFoundError.new("No route for #{req.method} #{req.path} #{Puppet::Network::HTTP::NOT_FOUND_ERROR_MESSAGE}",
+                                                                  Puppet::Network::HTTP::Issues::HANDLER_NOT_FOUND)
```

Comment:
I think @rlinehan has it right, this error message should only be displayed to agents using old URL's that were previous valid, but are not in later versions.

There should be a generic error message for generic resource not found errors.  If there isn't a generic 404 handler, then at the very least the specific upgrade helper message should fire only when an agent that needs to be upgraded is making requests.